src/main/java/teammates/ui/template/FeedbackSessionsAdditionalSettingsFormSegment.java

```diff
@@ -0,0 +1,258 @@
+        additionalSettings.setResponseVisibleDateDisabled(true);
+        additionalSettings.setResponseVisibleTimeOptions(data.getTimeOptionsAsElementTags(null));
+        additionalSettings.setResponseVisibleImmediatelyChecked(false);
+        additionalSettings.setResponseVisiblePublishManuallyChecked(true);
+        additionalSettings.setResponseVisibleNeverChecked(false);
+        
+        
+        additionalSettings.setSendClosingEmailChecked(true);
+        additionalSettings.setSendOpeningEmailChecked(true);
+        additionalSettings.setSendPublishedEmailChecked(true);
+        
+        return additionalSettings;
+    }
+    
+    public static FeedbackSessionsAdditionalSettingsFormSegment getFormSegmentWithExistingValues(
+                                                                    PageData data, 
+                                                                    FeedbackSessionAttributes feedbackSession) {
+        FeedbackSessionsAdditionalSettingsFormSegment additionalSettings = new FeedbackSessionsAdditionalSettingsFormSegment();
+        
+        setSessionVisibleSettings(data, feedbackSession, additionalSettings);
+        setResponseVisibleSettings(data, feedbackSession, additionalSettings);
+        setEmailSettings(feedbackSession, additionalSettings);
+        
+        return additionalSettings;
+    }
+
+    private static void setEmailSettings(FeedbackSessionAttributes feedbackSession,
+                                         FeedbackSessionsAdditionalSettingsFormSegment additionalSettings) {
+        additionalSettings.setSendClosingEmailChecked(feedbackSession.isClosingEmailEnabled);
+        additionalSettings.setSendOpeningEmailChecked(feedbackSession.isOpeningEmailEnabled);
+        additionalSettings.setSendPublishedEmailChecked(feedbackSession.isPublishedEmailEnabled);
+    }
+
+    private static void setResponseVisibleSettings(PageData data,
+                                                   FeedbackSessionAttributes feedbackSession,
+                                                   FeedbackSessionsAdditionalSettingsFormSegment additionalSettings) {
+        boolean hasResultVisibleDate = !TimeHelper.isSpecialTime(feedbackSession.resultsVisibleFromTime);
+        
+        additionalSettings.setResponseVisibleDateChecked(hasResultVisibleDate);
+        additionalSettings.setResponseVisibleDateValue(hasResultVisibleDate ? 
+                                                       TimeHelper.formatDate(feedbackSession.resultsVisibleFromTime) :
+                                                       "");
+        additionalSettings.setResponseVisibleTimeOptions(
+                                        data.getTimeOptionsAsElementTags(
+                                                            hasResultVisibleDate ?
+                                                            feedbackSession.resultsVisibleFromTime :
+                                                            null));
```

Comment:
According to the Java standard, this would be preferable:
```
additionalSettings.setResponseVisibleTimeOptions(
                                        data.getTimeOptionsAsElementTags(
                                                            hasResultVisibleDate
                                                            ? feedbackSession.resultsVisibleFromTime
                                                            : null));
```